test/testcafe/spec/uiDemo_spec.js

```diff
@@ -0,0 +1,39 @@
+import { RequestMock } from 'testcafe';
+import UiDemoPageObject from '../framework/page-objects/UiDemoPageObject';
+import uiDemoResponse from '../../../playground/mocks/data/idp/idx/_ui-demo.json';
+
+const uiDemoMock = RequestMock()
+  .onRequestTo('http://localhost:3000/idp/idx/introspect')
+  .respond(uiDemoResponse);
+
+async function setup(t) {
+  const pageObject = new UiDemoPageObject(t);
+  await pageObject.navigateToPage();
+  await t.expect(pageObject.formExists()).eql(true);
+  return pageObject;
+}
+
+fixture('UI demo');
+
+test
+  .requestHooks(uiDemoMock)('UI demo VRT', async t => {
+    const pageObject = await setup(t);
+
+    // freeze the spinner element so screenshots are consistent
+    await pageObject.stopSpinnerAnimation();
+
+    await t.customActions.compareScreenshot('one');
+    await pageObject.scrollToNextPage();
+
+    // possible bug in testcafe where the animated progress bar at the bottom is displaying after scrolling
+    // causing inconsistent screenshots
+    await pageObject.hideProgressBar();
```

Comment:
this was a bug in testcafe that was found previously and fixed: https://github.com/DevExpress/testcafe/issues/2076

but the bug persists after scrolling the page down for some reason